docs/protocols/xlang_object_graph_spec.md

```diff
@@ -0,0 +1,657 @@
+| NULL FLAG           | `-3`       | This flag indicates the object is a null value. We don't use another byte to indicate REF, so that we can save one byte.                                |
+| REF FLAG            | `-2`       | This flag indicates the object is already serialized previously, and fury will write a ref id with unsigned varint format instead of serialize it again |
+| NOT_NULL VALUE FLAG | `-1`       | This flag indicates the object is a non-null value and fury doesn't track ref for this type of object.                                                  |
+| REF VALUE FLAG      | `0`        | This flag indicates the object is referencable and the first time to serialize.                                                                         |
+
+When reference tracking is disabled globally or for specific types, or for certain types within a particular
+context(e.g., a field of a type), only the `NULL` and `NOT_NULL VALUE` flags will be used for reference meta.
+
+For languages which doesn't support reference such as rust, reference tracking must be disabled for correct
+deserialization by fury rust implementation.
+
+For languages whose object values are not null by default:
+
+- In rust, Fury takes `Option:None` as a null value
+- In c++, Fury takes `std::nullopt` as a null value
+- In golang, Fury takes `null interface/pointer` as a null value
+
+If one want to deserialize in languages like `Java/Python/JavaScript`, he should mark the type with all fields
+not-null by default, or using schema-evolution mode to carry the not-null fields info in the data.
+
+## Type Meta
+
+For every type to be serialized, it must be registered with an optional ID first. The registered type will have a
+user-provided or an auto-growing unsigned int i.e. `type_id`. The registration can be used for security check and type
+identification. The id of user registered type will be added by `64` to make space for Fury internal data types.
+
+Depending on whether meta share mode and registration is enabled for current type, Fury will write type meta
+differently.
+
+### Schema consistent
+
+If schema consistent mode is enabled globally or enabled for current type, type meta will be written as a fury unsigned
+varint of `type_id`.
+
+### Schema evolution
+
+If schema evolution mode is enabled globally or enabled for current type, type meta will be written as follows:
+
+- If meta share mode is not enabled, type meta will be written as schema consistent mode. Additionally, field meta such
+  as field type and name will be written with the field value using a key-value like layout.
+- If meta share mode is enabled, type meta will be written as a meta-share encoded binary if type hasn't been written
+  before, otherwise an unsigned varint id which references to previous written type meta will be written.
+
+## Meta share
```

Comment:
We can explain the `Meta share` here，including its usage and the conditions under which it is used.

Reply:
I updated the meta spec, could you review it again.